treeherder/etl/seta.py

```diff
@@ -0,0 +1,210 @@
+import logging
+
+from treeherder.seta.runnable_jobs import RunnableJobsClient
+
+LOG = logging.getLogger(__name__)
+
+
+class Treecodes:
+    """This class contain all the mapping we need in SETA and makes it works"""
+
+    def __init__(self, repo_name='mozilla-inbound'):
+        # default to query all jobs on mozilla-inbound branch
+        self.tbplnames = {}
+        self.jobtypes = []
+        self.jobnames = []
+
+        for job in RunnableJobsClient().query_runnable_jobs(repo_name)['results']:
+            # e.g. web-platform-tests-4
+            # e.g. Ubuntu VM 12.04 x64 mozilla-inbound opt test web-platform-tests-4 OR
+            #      desktop-test-linux64/opt-web-platform-tests-4
+            testtype, job_identifier = self._testtype_job_identifier(job)
+
+            # XXX: This has the danger of falling out of date
+            if not testtype or 'mulet' in job['platform']:
+                LOG.info('Ignoring {}'.format(job['ref_data_name']))
+                continue
+
+            # XXX: This has the danger of falling out of date
+            for i in ('dep', 'nightly', 'non-unified', 'valgrind', 'build'):
+                if testtype.find(i) != -1:
+                    LOG.info('Ignoring {}'.format(job['ref_data_name']))
+                    continue
+
+            platform = job['platform']
+            buildtype = job['platform_option']
+            self.jobtypes.append([platform, buildtype, testtype])
+
+            # It's about get jobnames and both buildbot and taskcluster job have '?'
+            # when the job_group_symbol is unknown.
+            self.jobnames.append(self._get_jobnames(job, testtype, job_identifier))
+
+    def _testtype_job_identifier(self, job):
+        '''
+        job - treeherder job
+        '''
+        job_identifier = None
+        testtype = None
+
+        # the testtype of builbot job can been found in 'ref_data_name'
+        # like web-platform-tests-4 in "Ubuntu VM 12.04 x64 mozilla-inbound
+        #  opt test web-platform-tests-4"
+        if job['build_system_type'] == 'buildbot':
+            testtype = job['ref_data_name'].split(' ')[-1]
+            job_identifier = job['ref_data_name']
+
+        # taskcluster's testtype is a part of its 'job_type_name' like reftest-2
+        # for [TC] Linux64 reftest-2
+        elif job['build_system_type'] == 'taskcluster':
+            # The test name on TaskCluster comes to a sort of combination
+            # (e.g desktop-test-linux64/debug-jittests-3) and asan job can
+            # been referenced as a opt job.
+            # NOTE: jobs like 'MacOSX64 Static Analysis Opt' will be left aside
+            # because they are not a test job.
+            # NOTE: The day we have Buildbot bridge in-tree we will have to support it
```

Comment:
can you file a bug about BBB and SETA.  Ideally SETA would run on all TC tasks- I would have to see what the BBB tasks look like and ensure we can apply existing SETA data to them.